packages/eslint-plugin/src/rules/no-shorthands.ts

```diff
@@ -0,0 +1,122 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/utils';
+import * as CSS from 'csstype';
+
+import { createRule } from '../utils/createRule';
+import { isIdentifier, isObjectExpression, isProperty } from '../utils/helpers';
+
+export const RULE_NAME = 'no-shorthands';
+
+// This collection is a map simply for faster access when checking if a CSS property is unsupported
+// @griffel/core has the same definition, but ESLint plugin should not depend on it
+const UNSUPPORTED_CSS_PROPERTIES: Record<keyof CSS.StandardShorthandProperties, true> = {
+  all: true,
+  animation: true,
+  background: true,
+  backgroundPosition: true,
+  border: true,
+  borderBlock: true,
+  borderBlockEnd: true,
+  borderBlockStart: true,
+  borderBottom: true,
+  borderColor: true,
+  borderImage: true,
+  borderInline: true,
+  borderInlineEnd: true,
+  borderInlineStart: true,
+  borderLeft: true,
+  borderRadius: true,
+  borderRight: true,
+  borderStyle: true,
+  borderTop: true,
+  borderWidth: true,
+  columns: true,
+  columnRule: true,
+  flex: true,
+  flexFlow: true,
+  font: true,
+  gap: true,
+  grid: true,
+  gridArea: true,
+  gridColumn: true,
+  gridRow: true,
+  gridTemplate: true,
+  lineClamp: true,
+  listStyle: true,
+  margin: true,
+  mask: true,
+  maskBorder: true,
+  motion: true,
+  offset: true,
+  outline: true,
+  overflow: true,
+  overscrollBehavior: true,
+  padding: true,
+  placeItems: true,
+  placeSelf: true,
+  textDecoration: true,
+  textEmphasis: true,
+  transition: true,
+};
+
+function findShorthandProperties(
+  node: TSESTree.ObjectExpression,
+  isRoot = false,
+  result: TSESTree.Identifier[] = [],
+): TSESTree.Identifier[] {
+  for (const propertyNode of node.properties) {
+    if (isProperty(propertyNode)) {
+      if (isIdentifier(propertyNode.key) && !isRoot) {
+        if (Object.prototype.hasOwnProperty.call(UNSUPPORTED_CSS_PROPERTIES, propertyNode.key.name)) {
```

Comment:
nit: these nested if blocks can just be reduced to a single one right ?

Reply:
Yes, but IMO it's more readable